IsADirectoryError: [Errno 21] Is a directory: '/Users/<user>/MobSF/uploads/<hash>/Pods/SQLite.swift'

## ENVIRONMENT

```
OS and Version: MacOS 10.15.6
Python Version: Python 3.8.5
MobSF Version: Mobile Security Framework v3.1.4 Beta
```

## EXPLANATION OF THE ISSUE

Uploaded ZIP file of iOS source code and stopped scanning after throwing following error in the UI:
```
Error
Inappropriate argument type.

'NoneType' object is not subscriptable
```

## STEPS TO REPRODUCE THE ISSUE

```
1. ZIP iOS source code
2. Upload to MobSF UI
3. Returns error while scanning
```

## LOG FILE

```
[INFO] 08/Sep/2020 10:04:29 - OS: Darwin
[INFO] 08/Sep/2020 10:04:29 - Platform: macOS-10.15.6-x86_64-i386-64bit
[INFO] 08/Sep/2020 10:04:29 - Dist: darwin 19.6.0 
[INFO] 08/Sep/2020 10:04:29 - MobSF Basic Environment Check
[INFO] 08/Sep/2020 10:04:29 - Checking for Update.
[INFO] 08/Sep/2020 10:04:29 - No updates available.
[INFO] 08/Sep/2020 10:04:32 - Starting Analysis on : Archive.zip
[INFO] 08/Sep/2020 10:04:32 - Extracting ZIP
[INFO] 08/Sep/2020 10:04:32 - Unzipping
[INFO] 08/Sep/2020 10:04:41 - Checking for ZIP Validity and Mode
[INFO] 08/Sep/2020 10:04:41 - Redirecting to iOS Source Code Analyzer
[INFO] 08/Sep/2020 10:04:41 - iOS Static Analysis Started
[INFO] 08/Sep/2020 10:04:41 - iOS Source Code Analysis Started
[INFO] 08/Sep/2020 10:04:41 - ZIP Already Extracted
[INFO] 08/Sep/2020 10:04:41 - Generating Hashes
[INFO] 08/Sep/2020 10:04:42 - Get Files, BIN Plist -> XML, and Normalize
[INFO] 08/Sep/2020 10:04:43 - iOS Info.plist Analysis Started
[INFO] 08/Sep/2020 10:04:43 - Finding Info.plist in iOS Source
[INFO] 08/Sep/2020 10:04:43 - Checking Permissions
[INFO] 08/Sep/2020 10:04:43 - Checking for Insecure Connections
[INFO] 08/Sep/2020 10:04:43 - Fetching Details from App Store: $(PRODUCT_BUNDLE_IDENTIFIER)
[WARNING] 08/Sep/2020 10:04:43 - Unable to get app details.
[INFO] 08/Sep/2020 10:04:43 - Starting iOS Source Code and PLIST Analysis
[ERROR] 08/Sep/2020 10:05:04 - iOS Source Code Analysis
Traceback (most recent call last):
  File "/Users/<user>/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/code_analysis.py", line 72, in ios_source_analysis
    pfile.read_text('utf-8', 'ignore'), relative_java_path)
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pathlib.py", line 1232, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pathlib.py", line 1218, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
IsADirectoryError: [Errno 21] Is a directory: '/Users/<user>/Mobile-Security-Framework-MobSF/uploads/c7e6bb61c08833d2a8fd2896a75edc3f/Pods/SQLite.swift'
[INFO] 08/Sep/2020 10:05:04 - Fetching icon path
[ERROR] 08/Sep/2020 10:05:04 - Error Perfroming Static Analysis
Traceback (most recent call last):
  File "/Users/<user>/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/static_analyzer.py", line 199, in static_analyzer_ios
    list(set(code_analysis_dic['urls_list'])))
TypeError: 'NoneType' object is not subscriptable
[ERROR] 08/Sep/2020 10:05:04 - 'NoneType' object is not subscriptable
[ERROR] 08/Sep/2020 10:05:04 - Internal Server Error: /StaticAnalyzer_iOS/
```

--- a/StaticAnalyzer/views/ios/code_analysis.py
+++ b/StaticAnalyzer/views/ios/code_analysis.py
@@ -65,7 +65,8 @@ def ios_source_analysis(src):
             if (
                 (pfile.suffix in ('.m', '.swift')
                     and any(skip_path in pfile.as_posix()
-                            for skip_path in skp) is False)
+                            for skip_path in skp) is False
+                    and pfile.is_dir() is False)
             ):
                 relative_java_path = pfile.as_posix().replace(src, '')
                 urls, urls_nf, emails_nf = url_n_email_extract(

